src/sync/partial_sync.cpp

```diff
@@ -241,32 +335,95 @@ RowExpr write_subscription(std::string const& object_type, std::string const& na
     if (columns.matches_property == npos) {
         auto target_table = ObjectStore::table_for_object_type(group, object_type);
         columns.matches_property = table->add_column_link(type_LinkList, matches_property, *target_table);
-    } else {
+    }
+    else {
         // FIXME: Validate that the column type and link target are correct.
     }
 
-    size_t row_ndx = validate_existing_subscription(*table, columns, name, query, matches_property);
-    if (row_ndx == npos) {
+    // Find existing subscription (if any)
+    auto row_ndx = table->find_first_string(columns.name, name);
+    if (row_ndx != npos) {
+        // If an subscription exist, we only update the query and TTL if allowed to.
+        // TODO: Consider how Binding API's are going to use this. It might make sense to disallow
+        // updating TTL using this API and instead require updates to TTL to go through a managed Subscription.
+        if (update) {
+            // Check that we don't replace the existing query with a query on a new type.
+            // There is nothing that prevents Sync from handling this, but allowing it
+            // will complicate Binding API's, so for now it is disallowed.
+            auto existing_matching_property = table->get_string(columns.matches_property_name, row_ndx);
+            if (existing_matching_property != matches_property) {
```

Comment:
This check exists in both arms of `if (update)`, so it should just be pulled out of the `if`. The error message also should include the name of the subscription.